ample/util/options_processor.py

```diff
@@ -412,13 +406,12 @@ def process_mr_options(optd):
 
     # Model building programs
     if optd['refine_rebuild_arpwarp'] or optd['shelxe_rebuild_arpwarp']:
-        if not (os.environ.has_key('warpbin')
-                and os.path.isfile(os.path.join(os.environ['warpbin'], "auto_tracing.sh"))):
+        if not ('warpbin' in os.environ and os.path.isfile(os.path.join(os.environ['warpbin'], "auto_tracing.sh"))):
```

Comment:
Instead of doing the same `os.path.join` twice, I would re-write this bit like:

```python
if optd['refine_rebuild_arpwarp'] or optd['shelxe_rebuild_arpwarp']:
    auto_tracing_sh = os.path.join(os.environ['warpbin'], "auto_tracing.sh")
    if 'warpbin' in os.environ and os.path.isfile(auto_tracing_sh):
        logger.info('Using arpwarp script: %s', auto_tracing_sh)
    else:
        logger.warn('Cannot find arpwarp script! Disabling use of arpwarp.')
        optd['refine_rebuild_arpwarp'] = False
        optd['shelxe_rebuild_arpwarp'] = False
```

Reply:
Thinking about this, what happens if a user decides not to install arpwarp ... then the `warpbin` is not even in `os.environ` ... we need to be careful here

Reply:
Please include this as well Jens